climada/engine/forecast.py

```diff
@@ -0,0 +1,897 @@
+
+    def ei_exp(self, run_datetime=None):
+        """
+        Expected annual impact
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        Returns
+        -------
+        float
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
+        return self._impact[haz_ind].eai_exp
+
+    def ai_agg(self, run_datetime=None):
+        """ Annual average impact
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        Returns
+        -------
+        float
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
+        return self._impact[haz_ind].aai_agg
+
+    def haz_summary_str(self, run_datetime=None):
+        """ provide a summary string for the hazard part of the forecast
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        Returns
+        -------
+        str
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
+        return (self.hazard[haz_ind].tag.haz_type +
+                '_' +
+                self.haz_model +
+                '_run' +
+                run_datetime.strftime('%Y%m%d%H') +
+                '_event' +
+                self.event_date.strftime('%Y%m%d')
+                )
+
+    def summary_str(self, run_datetime=None):
+        """ provide a summary string for the impact forecast
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        Returns
+        -------
+        str
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        return (self.haz_summary_str(run_datetime) +
+                '_' +
+                self.exposure_name
+                )
+
+    def lead_time(self, run_datetime=None):
+        """ provide the lead time for the impact forecast
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            select the used hazard by the run_datetime.
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        return self.event_date - run_datetime
+
+    def calc(self, force_reassign=False, check_plot=False):
+        """ calculate the impact using
+        exposure, hazard, and vulnerabilty.
+
+        Parameters
+        ----------
+        force_reassign: bool, optional
+            Reassign centroids.
```

Comment:
```suggestion
            Reassign hazard centroids to the exposure.
```
But to what hazard? `self.hazard` is a list of hazards... Nowhere is it said that they must have the same centroids... 

Note that in `impact.calc` the centroids are automatically assigned if not yet done. This is maybe more robust than the `force_reassign`.

Reply:
This was wrongly outside of the loop. It is now inside the loop. That way the centroids do not have to be the same. (And for the rare case, when there are two hazards provided with same Hazard.tag.haz_type but with different centroids. This flag makes sure that the exposure always reassigns the centroids, and does not misinterpret the previously assigned centroids with the same haz_type.)